### PR TITLE
Another pass at docs

### PR DESCRIPTION
**Pull request by edenlightning**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-lash/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/general/predictions.rst [hunk 1]
```diff
@@ -15,19 +15,20 @@ You can pass in a sample of data (image file path, a string of text, etc) to the
 	
 .. code-block:: python
 
-    from flash import Trainer
-    from flash import download_data
-    from flash.vision import ImageClassificationData, ImageClassifier
+	from flash import Trainer
```
Comment (carmocca) [inline docs/source/general/predictions.rst:18]: IDK if this matters but these code-blocks have more indentation than others

File: docs/source/general/training.rst [hunk 2]
```diff
@@ -71,7 +72,7 @@ Flash tasks supports many advanced training functionalities out-of-the-box, such
     # Train on TPUs
     trainer.fit(tpu_cores=8)
 
-You can add to the flash Trainer any argument from the Lightning trainer! Learn more about the Lightning Trainer `here <https://pytorch-lightning.readthedocs.io/en/stable/trainer.html>`.
+You can add to the flahs Trainer any argument from the Lightning trainer! Learn more about the Lightning Trainer `here <https://pytorch-lightning.readthedocs.io/en/stable/trainer.html>`_.
```
Comment (carmocca) [inline docs/source/general/training.rst:75]:
```suggestion
You can add to the flash Trainer any argument from the Lightning trainer! Learn more about the Lightning Trainer `here <https://pytorch-lightning.readthedocs.io/en/stable/trainer.html>`_.
```

File: docs/source/general/training.rst [hunk 3]
```diff
@@ -2,12 +2,13 @@
 Training from scratch
 *********************
 
-Some Flash tasks have been pretrained on large data sets, to accelerate your training (calling the :func:`~flash.Trainer.finetune` method using a pretrained backbone will fine-tune the backbone to generate a model customized to your data set and desired task). If you want to train the task from scratch instead, pass `pretrained=False` parameter when creating your task. Then, use the :func:`~flash.Trainer.fit` method to train your model.
+Some Flash tasks have been pretrained on large data sets, to accelorate your training (calling the :func:`~flash.core.trainer.Trainer.finetune` method using a pretrained backbone will fine-tune the backbone to generate a model customized to your data set and desired task). If you want to train the task from scratch instead, pass `pretrained=False` parameter when creating your task. Then, use the :func:`~flash.core.trainer.Trainer.fit` method to train your model.
```
Comment (carmocca) [inline docs/source/general/training.rst:5]:
```suggestion
Some Flash tasks have been pretrained on large data sets. To accelerate your training, calling the :func:`~flash.core.trainer.Trainer.finetune` method using a pretrained backbone will fine-tune the backbone to generate a model customized to your data set and desired task. If you want to train the task from scratch instead, pass `pretrained=False` parameter when creating your task. Then, use the :func:`~flash.core.trainer.Trainer.fit` method to train your model.
```

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -1,35 +1,48 @@
 Quick Start
 ===========
-Flash is a high-level deep learning framework for fast prototyping, finetuning and solving applied deep learning problems.
+
+Flash is a high-level deep learning framework for fast prototyping, baselining, finetuning and solving deep learning problems. It features a collaction of tasks for you to use for inference and finetuning out of the box, and an easy to implement API to customuzie every step of the process.
```
Comment (carmocca) [inline docs/source/quickstart.rst:4]:
```suggestion
Flash is a high-level deep learning framework for fast prototyping, baselining, finetuning and solving deep learning problems. It features a set of tasks for you to use for inference and finetuning out of the box, and an easy to implement API to customize every step of the process.
```

File: docs/source/quickstart.rst [hunk 5]
```diff
@@ -158,38 +171,55 @@ Once your model is finetuned, use it for prediction anywhere you want!
...
+              metrics=metrics,
+              learning_rate=learning_rate,
+          )
+
+          self.save_hyperparameters()
+
+          self.linear = torch.nn.Linear(num_inputs, num_classes)
+
+        def forward(self, x):
+            return self.linear(x)
```
Comment (teddykoker) [inline docs/source/quickstart.rst:222]: This is part of the "create your own task", might want to remove this for now and replace once we have something good

File: docs/source/quickstart.rst [hunk 6]
```diff
@@ -1,35 +1,48 @@
...
+^^^^^^^^^^^
+Flash is built on top of `Pytorch Lightning
+<https://github.com/PyTorchLightning/pytorch-lightning>`_, a powerful deep learning research framework for training models at scale. With the power of Lightning, you can train your flash tasks on any hardware: CPUs, GPUs or TPUs without any code changes. 
 
-Unlike other high-level frameworks, Flash provides full flexibility, all tasks can be customized, overidden and extended using Lightning.
+Flexibility
+^^^^^^^^^^^
+Unlike other high-level frameworks, it's easy to customize the Flash tasks with `Pytorch Lightning
+<https://github.com/PyTorchLightning/pytorch-lightning>`_ components to get all the flexibility you need. Lightning is just
+organized PyTorch with unecessary engineering details abstracted away.
```
Comment (carmocca) [inline docs/source/quickstart.rst:40]:
```suggestion
organized PyTorch with the unecessary engineering details abstracted away.
```

**Review by teddykoker — CHANGES_REQUESTED** [review]

File: docs/source/general/predictions.rst (hunk 1 above)
Comment (teddykoker) [reply docs/source/general/predictions.rst:18]: yeah this was an issue I had to remove previously

File: docs/source/general/training.rst [hunk 7]
```diff
@@ -2,12 +2,13 @@
...
+
 
 .. code-block:: python
 
     import flash
-    from flash import download_data
+    from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/general/training.rst:11]:
```suggestion
    from flash import download_data
```

File: docs/source/reference/image_classification.rst [hunk 8]
```diff
@@ -16,46 +16,36 @@ The task of identifying what is in an image is called `image classification <htt
...
-Use the :class:`~flash.text.ImageClassificatier` pretrained model for inference on any string sequence using :func:`~flash.text.TextClassifier.predict`:
+Use the :class:`~flash.vision.ImageClassifier` pretrained model for inference on any string sequence using :func:`~flash.vision.ImageClassifier.predict`:
 
 .. code-block:: python
 
-	# import our libraries
-	from flash.text import TextClassifier
+    # import our libraries
+    from flash import Trainer
+    from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/reference/image_classification.rst:28]:
```suggestion
    from flash import download_data
```

File: docs/source/reference/image_classification.rst [hunk 9]
```diff
@@ -97,16 +87,16 @@ Now all we need is three lines of code to build to train our task!
 
 .. code-block:: python
 
-	import flash
-	from flash import download_data
-	from flash.vision import ImageClassificationData, ImageClassifier
+    import flash
+    from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/reference/image_classification.rst:91]:
```suggestion
    from flash import download_data
```

File: docs/source/reference/tabular_classification.rst [hunk 10]
```diff
@@ -42,70 +42,72 @@ We can create :class:`~flash.tabular.TabularData` from csv files using the :func
...
 
-.. code-block:: python
+.. code-block::
 
-    from pytorch_lightning.metrics.classification import Accuracy, Precision, Recall
-    import flash
-    from flash import download_data
-    from flash.tabular import TabularClassifier, TabularData
+  import flash
+  from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/reference/tabular_classification.rst:50]:
```suggestion
  from flash import download_data
```

File: docs/source/reference/tabular_classification.rst [hunk 11]
```diff
@@ -42,70 +42,72 @@ We can create :class:`~flash.tabular.TabularData` from csv files using the :func
...
-You can make predictions on a pretrained model, that has already been trained for the titanic task:
+You can make predcitions on a pretrained model, that has already been trained for the titanic task:
 
 .. code-block:: python
 
 
-    from flash import download_data
-    from flash.tabular import TabularClassifier
-
+  from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/reference/tabular_classification.rst:97]:
```suggestion
  from flash import download_data
```

File: docs/source/reference/task.rst [hunk 12]
```diff
@@ -14,7 +14,7 @@ Example: Image Classification
 
 .. code-block:: python
 
-    from flash.model import Task
+    from flash.core.model import Task
```
Comment (teddykoker) [inline docs/source/reference/task.rst:17]: Task can be imported from root
```suggestion
    from flash import Task
```

File: docs/source/reference/text_classification.rst [hunk 13]
```diff
@@ -16,48 +16,41 @@ Text classification is the task of assigning a piece of text (word, sentence or
...
-Use the :class:`~flash.text.TextClassificatier` pretrained model for inference on any string sequence using :func:`~flash.text.TextClassifier.predict`:
+Use the :class:`~flash.text.classification.model.TextClassifier` pretrained model for inference on any string sequence using :func:`~flash.text.classification.model.TextClassifier.predict`:
 
 .. code-block:: python
 
-	# import our libraries
-	from flash.text import TextClassifier
+    from pytorch_lightning import Trainer
 
+    from flash.core.data import download_data
```
Comment (teddykoker) [inline docs/source/reference/text_classification.rst:27]:
```suggestion
    from flash import download_data
```

**Review by teddykoker — CHANGES_REQUESTED** [review]

**Review by teddykoker — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: docs/source/general/predictions.rst [hunk 14]
```diff
@@ -36,45 +37,19 @@ Predict on a csv file
...
-    from flash import download_data
-    from flash.vision import ImageClassificationData, ImageClassifier
-
-
-    # 1. Download the data set
-    download_data("https://pl-flash-data.s3.amazonaws.com/hymenoptera_data.zip", 'data/')
-
-    # 2. Load the model from a checkpoint
-    model = ImageClassifier.load_from_checkpoint("https://flash-weights.s3.amazonaws.com/image_classification_model.pt")
+	from flash.core.data import download_data
```
Comment (Borda) [inline docs/source/general/predictions.rst:40]: does this formating make any difference?

File: docs/source/quickstart.rst [hunk 15]
```diff
@@ -39,7 +52,7 @@ When you need more flexibility you can build your own tasks or simply use Lightn
 Install
 -------
 
-You can install flash using pip/conda:
+You can install flash using `pip/conda <https://pypi.org/project/lightning-flash/0.1.0rc2/>`_:
```
Comment (Borda) [inline docs/source/quickstart.rst:55]:
```suggestion
You can install flash using `pip install lightning-flash -U -pre`_:
```

File: docs/source/reference/tabular_classification.rst [hunk 16]
```diff
@@ -116,30 +118,30 @@ Or you can finetune your own model and use that for prediction:
...
 
-    # 6. Test model
-    trainer.test()
+  # 6. Test model
+  trainer.test()
 
-    predictions = model.predict("data/titanic/titanic.csv")
-    print(predictions)
+  predictions = model.predict("data/titanic/titanic.csv")
+  print(predictions)
```
Comment (Borda) [inline docs/source/reference/tabular_classification.rst:144]: add spaces

**Review by Borda — CHANGES_REQUESTED** [review]

File: docs/source/general/predictions.rst (hunk 14 above)
Comment (edenlightning) [reply docs/source/general/predictions.rst:40]: cleaned it

File: docs/source/reference/tabular_classification.rst (hunk 16 above)
Comment (edenlightning) [reply docs/source/reference/tabular_classification.rst:144]: where? mind adding suggestions?

**Comment by edenlightning** [comment]
@teddykoker @tchaton please take another look, modified finetuning

**Review by tchaton — APPROVED** [review]

**Review by teddykoker — APPROVED** [review]

**Comment by teddykoker** [comment]
Any idea why tests are failing @edenlightning ?

**Pull request merged**